src/weatherDataSource.js

```diff
@@ -0,0 +1,43 @@
+export default class WeatherDataSource {
+    constructor() {
+        this.apiKey = '84af29f03ef35bb4dd7f1e9b0ad575e2';
+    }  
+    getWeather(city) { 
+        fetch('https://api.openweathermap.org/data/2.5/weather?q=' + city + '&appid=' + this.apiKey)  
+            .then(function(resp) { 
+                return resp.json() })
+            .then(function(data) {
+                console.log(data);
+                //showWeather(data);
+                return data;
+            })
+            .catch(function(err) {
+                console.log(err);
+            });
+    }
```

Comment:
Ta funkcja wciąż nic nie zwraca ;) 
Jeśli to testowałaś, to prawdopodobnie sama przed sobą ukryłaś ten fakt przez linijkę console.log(data) - bo istotnie, wypisujesz dane do konsoli... ale returny mają tylko Twoje anonimowe funkcje wywoływane w .then().

Spróbuj sobie przetestować, co pojawia się w konsoli, jeśli usuniesz console.log(data) z Twojej klasy, a w index.js przetestujesz funkcjonalność Twojej klasy następującym kawałkiem kodu:

```
const weatherSource = new WeatherDataSource();

async function updateWeather() {
    const weather = await weatherSource.getWeather("Przemyśl");
    console.log("I used WeatherDataSource class and all I got was this: ", weather);
}

updateWeather();
```